extensions/shared/track-gutenberg-search.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { registerPlugin } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies
+ */
+import analytics from '../../_inc/client/lib/analytics';
+
+const trackGutenbergSearch = event => {
+	const searchTerm = event.target.value;
+	const hasResults =
+		document.getElementsByClassName( 'block-editor-inserter__no-results' ).length === 0;
+
+	if ( searchTerm.length < 3 ) {
+		return;
+	}
+
+	analytics.tracks.recordEvent( 'jetpack_gutenberg_block_picker_search_term', {
+		searchTerm: searchTerm,
+	} );
+
+	if ( hasResults ) {
+		return;
+	}
+
+	// Create a separate event for search with no results to make it easier to filter by them
+	analytics.tracks.recordEvent( 'jetpack_gutenberg_block_picker_no_results', {
+		searchTerm: searchTerm,
+	} );
+};
+
+registerPlugin( 'track-no-search-results', {
```

Comment:
I don't understand why it's called "no search results". Am I missing something?

Reply:
It started off by tracking only the case where we get no results, but now we track both cases!